dvc/repo/get.py

```diff
@@ -42,37 +41,47 @@ def get(url, path, out=None, rev=None):
     # and won't work with reflink/hardlink.
     dpath = os.path.dirname(os.path.abspath(out))
     tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
+    repo_dir = None
     try:
-        with external_repo(cache_dir=tmp_dir, url=url, rev=rev) as repo:
-            # Try any links possible to avoid data duplication.
-            #
-            # Not using symlink, because we need to remove cache after we are
-            # done, and to make that work we would have to copy data over
-            # anyway before removing the cache, so we might just copy it
-            # right away.
-            #
-            # Also, we can't use theoretical "move" link type here, because
-            # the same cache file might be used a few times in a directory.
-            repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
-
-            try:
-                output = repo.find_out_by_relpath(path)
-            except OutputNotFoundError:
-                output = None
-
-            if output and output.use_cache:
-                _get_cached(repo, output, out)
-            else:
+        try:
+            with external_repo(cache_dir=tmp_dir, url=url, rev=rev) as repo:
+                # Try any links possible to avoid data duplication.
+                #
+                # Not using symlink, because we need to remove cache after we
+                # are done, and to make that work we would have to copy data
+                # over anyway before removing the cache, so we might just copy
+                # it right away.
+                #
+                # Also, we can't use theoretical "move" link type here, because
+                # the same cache file might be used a few times in a directory.
+                repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
+
+                try:
+                    output = repo.find_out_by_relpath(path)
+                except OutputNotFoundError:
+                    output = None
+
+                if output and output.use_cache:
+                    _get_cached(repo, output, out)
+                    return
+
                 # Either an uncached out with absolute path or a user error
-                if os.path.isabs(path):
-                    raise FileNotFoundError
 
-                fs_copy(os.path.join(repo.root_dir, path), out)
+                repo_dir = repo.root_dir
+
+        except NotDvcRepoError:
+            # Not a DVC repository, continue below and copy from git
+            pass
+
+        if os.path.isabs(path):
+            raise FileNotFoundError
+
+        if not repo_dir:
+            repo_dir = cached_clone(url, rev=rev)
 
+        fs_copy(os.path.join(repo_dir, path), out)
     except (OutputNotFoundError, FileNotFoundError):
         raise PathMissingError(path, url)
-    except NotDvcRepoError:
-        raise UrlNotDvcRepoError(url)
```

Comment:
I think we should revert this at least for api code.

Reply:
@Suor Not sure I follow, `get` is not even part of the api.

Reply:
This is making better error message for code using external_repo, but that exception is caught in get and import code, so these two lines won't fire for them, but it can still fire for any other external_repo user like api.

Reply:
@Suor Yes, but we are looking at `get` here, not `external_repo`, which wasn't wrapped anyways. @skshetry has noticed that too today and will be adding a wrapper for `api.get_url` soon too.

Reply:
Disregarding whether we have or don't have a wrapper for api, which has functions beside `get_url`, `external_repo()` should raise errors with appropriate messages. Otherwise we force very tricky interface on all `external_repo()` users they need to catch some exceptions, and if they forget too than users will get wrong error messages. 

Removing this lines make `external_repo()` abstraction broken. We should restore it and we won't need any wrapper in api code and make using `external_repo()` easier.

Reply:
@Suor Once again, you are talking about a general problem under a PR that added support for specific feature that doesn't affect the general problem. There are no action points in this PR, let's stop flooding it with off-topic stuff please.